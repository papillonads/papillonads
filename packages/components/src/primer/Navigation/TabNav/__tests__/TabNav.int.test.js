import {
  defaultItems as items,
  regular,
  withButtonActions,
  withTextActions,
  withLinkActions,
  withNavigation,
  withMixedEnabledVisibleItems,
  inactive,
} from './TabNav.int.story'

jest.mock('@papillonads/library/array', () => ({
  getIndexItems: () => {},
  getIndexItemsWithSelected: () => {},
}))

const libraryArrayMockObject = require('@papillonads/library/array')

describe('<TabNav />', () => {
  const indexItemsDataObject = [
    { ...items[0], index: 0 },
    { ...items[1], index: 1 },
  ]

  const newIndexItemsWithSelectedDataObject = [
    { ...items[0], index: 0, isSelected: true, enabled: true, visible: true },
    { ...items[1], index: 1, isSelected: false, enabled: true, visible: true },
  ]

  function mockLibraryForRegularItems() {
    jest.spyOn(libraryArrayMockObject, 'getIndexItems').mockReturnValue(indexItemsDataObject)

    jest.spyOn(libraryArrayMockObject, 'getIndexItemsWithSelected').mockReturnValue(newIndexItemsWithSelectedDataObject)
  }

  afterEach(() => jest.clearAllMocks())

  describe('Render', () => {
    test('must match regular()', () => {
      mockLibraryForRegularItems()
      expect(global.renderToJSON(regular())).toMatchSnapshot()
    })

    test('must match withButtonActions()', () => {
      mockLibraryForRegularItems()
      expect(global.renderToJSON(withButtonActions())).toMatchSnapshot()
    })

    test('must match withTextActions()', () => {
      mockLibraryForRegularItems()
      expect(global.renderToJSON(withTextActions())).toMatchSnapshot()
    })

    test('must match withLinkActions()', () => {
      mockLibraryForRegularItems()
      expect(global.renderToJSON(withLinkActions())).toMatchSnapshot()
    })

    test('must match withNavigation()', () => {
      mockLibraryForRegularItems()
      expect(global.renderToJSON(withNavigation())).toMatchSnapshot()
    })

    test('must match withMixedEnabledVisibleItems()', () => {
      mockLibraryForRegularItems()
      expect(global.renderToJSON(withMixedEnabledVisibleItems())).toMatchSnapshot()
    })

    test('must match inactive()', () => {
      mockLibraryForRegularItems()
      expect(global.renderToJSON(inactive())).toMatchSnapshot()
    })
  })
})
