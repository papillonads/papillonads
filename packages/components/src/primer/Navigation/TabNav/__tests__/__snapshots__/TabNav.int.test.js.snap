// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabNav /> Render must match inactive() 1`] = `
<div>
  <div
    class="tabnav tabnav-inactive"
  >
    <nav
      aria-label="Foo bar"
      class="tabnav-tabs"
    >
      <a
        aria-current="page"
        class="tabnav-tab"
        href="#url"
      >
        Foo tab
      </a>
      <a
        class="tabnav-tab"
        href="#url"
      >
        Bar tab
      </a>
    </nav>
  </div>
</div>
`;

exports[`<TabNav /> Render must match regular() 1`] = `
<div>
  <div
    class="tabnav"
  >
    <nav
      aria-label="Foo bar"
      class="tabnav-tabs"
    >
      <a
        aria-current="page"
        class="tabnav-tab"
        href="#url"
      >
        Foo tab
      </a>
      <a
        class="tabnav-tab"
        href="#url"
      >
        Bar tab
      </a>
    </nav>
  </div>
</div>
`;

exports[`<TabNav /> Render must match withButtonActions() 1`] = `
<div>
  <div
    class="tabnav"
  >
    <a
      class="btn btn-sm float-right"
      href="#url"
      role="button"
    >
      Button
    </a>
    <nav
      aria-label="Foo bar"
      class="tabnav-tabs"
    >
      <a
        aria-current="page"
        class="tabnav-tab"
        href="#url"
      >
        Foo tab
      </a>
      <a
        class="tabnav-tab"
        href="#url"
      >
        Bar tab
      </a>
    </nav>
  </div>
</div>
`;

exports[`<TabNav /> Render must match withLinkActions() 1`] = `
<div>
  <div
    class="tabnav"
  >
    <div
      class="float-right"
    >
      <a
        class="tabnav-extra"
        href="#url"
      >
        Tabnav extra link
      </a>
      <a
        class="tabnav-extra"
        href="#url"
      >
        Tabnav extra link
      </a>
    </div>
    <nav
      aria-label="Foo bar"
      class="tabnav-tabs"
    >
      <a
        aria-current="page"
        class="tabnav-tab"
        href="#url"
      >
        Foo tab
      </a>
      <a
        class="tabnav-tab"
        href="#url"
      >
        Bar tab
      </a>
    </nav>
  </div>
</div>
`;

exports[`<TabNav /> Render must match withTextActions() 1`] = `
<div>
  <div
    class="tabnav"
  >
    <div
      class="tabnav-extra float-right"
    >
      Tabnav widget text here.
    </div>
    <nav
      aria-label="Foo bar"
      class="tabnav-tabs"
    >
      <a
        aria-current="page"
        class="tabnav-tab"
        href="#url"
      >
        Foo tab
      </a>
      <a
        class="tabnav-tab"
        href="#url"
      >
        Bar tab
      </a>
    </nav>
  </div>
</div>
`;
